duration.go

```diff
@@ -0,0 +1,161 @@
+package null
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	errBadDurationSQLType = errors.New("incompatible SQL type for Scan() into null.Duration")
+)
+
+// Duration is a nullable time.Duration. It supports SQL and JSON serialization.
+// It will marshal to null if null.
+type Duration struct {
+	Duration time.Duration
+	Valid    bool
+}
+
+// Scan implements the Scanner interface.
+// Supports the ´interval` type for PostgreSQL
+func (d *Duration) Scan(value interface{}) error {
+	var source string
+	switch value.(type) {
+	case string:
+		source = value.(string)
+	case []byte:
+		source = string(value.([]byte))
+	default:
+		return errBadDurationSQLType
+	}
+
+	split := strings.Split(source, ":")
+	if len(split) != 3 {
+		return errBadDurationSQLType
+	}
+
+	hours, err := strconv.Atoi(split[0])
+	if err != nil {
+		return errBadDurationSQLType
+	}
+
+	minutes, err := strconv.Atoi(split[1])
+	if err != nil {
+		return errBadDurationSQLType
+	}
+
+	seconds, err := strconv.Atoi(split[2])
+	if err != nil {
+		return errBadDurationSQLType
+	}
+
+	duration := (time.Hour * time.Duration(hours)) +
+		(time.Minute * time.Duration(minutes)) +
+		(time.Second * time.Duration(seconds))
```

Comment:
What if `minutes` or `second` is higher than `60`? Not sure if it needs to be overly strict with this (especially since you can have `61` seconds with leap seconds), but would be good to at least at a test case for this.